### PR TITLE
[Backport] Add avro dependency to parquet extension

### PR DESCRIPTION
**Pull request by suneet-s**
Backport of apache/druid#9124 to 0.17.0-iap

### CODE REVIEW

File: extensions-core/parquet-extensions/pom.xml [hunk 1]
```diff
@@ -412,7 +400,7 @@
     <dependency>
       <groupId>org.apache.avro</groupId>
       <artifactId>avro</artifactId>
-      <scope>provided</scope>
+      <version>${avro.version}></version>
```
Comment (jon-wei) [inline extensions-core/parquet-extensions/pom.xml]: This has an extra `>`

Comment (suneet-s) [reply extensions-core/parquet-extensions/pom.xml]: 😢 I cherry-picked the wrong commit ugh so much for trying to rush this in.

**Comment by jon-wei** [comment]
We'll need to backport https://github.com/apache/druid/pull/9120 to 0.17.0-iap, the tests are failing on the  issue fixed by that

**Comment by jon-wei** [comment]
I opened the backport at https://github.com/implydata/druid/pull/29

**Pull request merged**